cookbooks/equilibrium_grain_size/plugins/reference_profile.h

```diff
@@ -0,0 +1,207 @@
+/*
+  Copyright (C) 2011 - 2018 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_adiabatic_conditions_reference_profile_h
+#define _aspect_adiabatic_conditions_reference_profile_h
+
+
+#include <aspect/adiabatic_conditions/interface.h>
+
+#include <deal.II/base/parsed_function.h>
+
+namespace aspect
+{
+  namespace AdiabaticConditions
+  {
+    using namespace dealii;
+
+    /**
+     * A model in which the adiabatic profile is
+     * calculated by solving the hydrostatic equations for
+     * pressure and temperature in depth.
+     * The gravity is assumed to be in depth direction
+     * and the composition is either given by the initial
+     * composition at reference points or computed
+     * as a reference depth-function.
+     * All material parameters are computed by the
+     * material model plugin. The surface conditions are
+     * either constant or changing over time as prescribed
+     * by an user-provided function.
+     */
+    template <int dim>
+    class ReferenceProfile : public Interface<dim>
```

Comment:
Is this a copy of a similar class from elsewhere? If so, can you describe what is different? I wonder whether we could find a way to achieve the same goal with less code than results from copying a class and its implementation.

Reply:
Yes, this is very similar to the `adiabatic_conditions/compute_profile.h`. The only difference is that we use a reference density from prem model below a certain depth defined by `uppermost_mantle_thickness` parameter. I tried to remove the code and use `compute_profile` as a base model, but I was not very successful given the required functions from the interface class and the fact that computation of density required parameters that we needed to define. It might be easier to add the changes from our adiabatic profile in the current `compute_profile` file, but I am not sure if it will be useful for others.
Maybe I can ask you more about this later today and find a solution.

Reply:
@gassmoeller suggested a way to remove this class and it works with the existing implementation of `compute_profile` and the modifications to the material model. Thank you for pointing that out!